Update create automation dialog

piitaya opened the pull request:
## Proposed change

### If the user has blueprints

![CleanShot 2023-02-01 at 10 45 05](https://user-images.githubusercontent.com/5878303/216008015-369893ae-f393-4c68-8e4c-d9a297f75d58.png)


### If the user doesn't have blueprints

![CleanShot 2023-02-01 at 10 45 28](https://user-images.githubusercontent.com/5878303/216008111-c3951037-507f-4c53-acbb-a1739ff7e8fe.png)

## Type of change

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration


<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

piitaya committed:
Update create automation dialog

bramkragten commented on src/panels/config/automation/dialog-new-automation.ts:
```diff
@@ -30,59 +52,121 @@ class DialogNewAutomation extends LitElement implements HassDialog {
+                    : mdiAccount}
+                ></ha-svg-icon>
+                ${blueprint.name}
+                <span slot="secondary">${blueprint.path}</span>
```
Should we show the path? Feels technical

ildar170975 replied:
In my practice, a path's part - a filename - is used as the ONLY blueprint's name since in 99% cases I am using yaml to maintain automations. So, I suggest to display this filename.

piitaya replied:
The file name is available in the blueprint page. This dialog is only for user that want to start a blueprint from UI.

ildar170975 commented:
If a user has 30 blueprints - then all of them will be listed?

piitaya commented:
Yep, a scroll bar is displayed if there are many blueprints 🙂

piitaya committed:
Improve blueprint source

piitaya commented:
I updated the screenshot the display the source instead of the path and I added a tip link at the bottom.
This PR also support author name if provided in the blueprint (needs a backend change for that point)

piitaya committed:
Fetch blueprint on dialog open

ildar170975 commented:
I believe that most of blueprints are made by a user himself. Then why to print this “be me”? Probably an author label should be printed only when a blueprint is not made by a user.

bramkragten commented on src/panels/config/automation/dialog-new-automation.ts:
```diff
@@ -1,26 +1,57 @@
 import { fireEvent } from "../../../common/dom/fire_event";
+import { shouldHandleRequestSelectedEvent } from "../../../common/mwc/handle-request-selected-event";
+import { stringCompare } from "../../../common/string/compare";
 import "../../../components/ha-blueprint-picker";
```
Unused?

bramkragten commented on src/panels/config/automation/dialog-new-automation.ts:
```diff
@@ -30,59 +61,152 @@ class DialogNewAutomation extends LitElement implements HassDialog {
-          </mwc-list-item>
+          </ha-list-item>
+          <li divider role="separator"></li>
+          ${processedBlueprints.length > 0 ? html`` : null}
```
```suggestion
```

piitaya committed:
Remove unused code
